ha/setup-cluster.sh

```diff
@@ -0,0 +1,188 @@
+#!/bin/bash
+
+set -x
+
+WORK_DIR="${1:-$(pwd)}"
+CONFIG_DIR="${2:-"${WORK_DIR}/config"}"
+
+CREATE_VM_SCRIPT="$(pwd)/create-vm.sh"
+
+VM01="node01"
+VM02="node02"
+VM03="node03"
+
+SSH="ssh -o StrictHostKeyChecking=no"
+SCP="scp -o StrictHostKeyChecking=no"
+
+check_requirements() {
+    hash virsh ssh-keygen wget virt-install qemu-img cloud-localds uuidgen || exit 127
+}
+
+create_nodes() {
+  for vm in "${VM01}" "${VM02}" "${VM03}"; do
+    ${CREATE_VM_SCRIPT} ${vm} $(pwd)
+  done
+  virsh list
+}
+
+get_nodes_ip_address() {
+  sleep 30
+  IP_VM01=$(virsh domifaddr ${VM01} | grep ipv4 | xargs | cut -d ' ' -f4 | cut -d '/' -f1)
+  IP_VM02=$(virsh domifaddr ${VM02} | grep ipv4 | xargs | cut -d ' ' -f4 | cut -d '/' -f1)
+  IP_VM03=$(virsh domifaddr ${VM03} | grep ipv4 | xargs | cut -d ' ' -f4 | cut -d '/' -f1)
+}
+
+write_hosts() {
+  cat > ${CONFIG_DIR}/hosts <<EOF
+127.0.0.1 localhost
+
+# The following lines are desirable for IPv6 capable hosts
+::1 ip6-localhost ip6-loopback
+fe00::0 ip6-localnet
+ff00::0 ip6-mcastprefix
+ff02::1 ip6-allnodes
+ff02::2 ip6-allrouters
+ff02::3 ip6-allhosts
+
+${IP_VM01} ${VM01}
+${IP_VM02} ${VM02}
+${IP_VM03} ${VM03}
+EOF
+}
+
+write_corosync_conf() {
+  cat > ${CONFIG_DIR}/corosync.conf <<EOF
+totem {
+        version: 2
+        secauth: off
+        cluster_name: testcluster
+        transport: udp
+}
+
+nodelist {
+        node {
+                ring0_addr: ${IP_VM01}
+                name: ${VM01}
+                nodeid: 1
+        }
+        node {
+                ring0_addr: ${IP_VM02}
+                name: ${VM02}
+                nodeid: 2
+        }
+        node {
+                ring0_addr: ${IP_VM03}
+                name: ${VM03}
+                nodeid: 3
+        }
+}
+
+quorum {
+        provider: corosync_votequorum
+        two_node: 0
+}
+
+qb {
+        ipc_type: native
+}
+
+logging {
+
+        fileline: on
+        to_stderr: on
+        to_logfile: yes
+        logfile: /var/log/corosync/corosync.log
+        to_syslog: no
+        debug: off
+}
+EOF
+}
+
+write_config_files() {
+  write_hosts
+  write_corosync_conf
+}
+
+run_in_all_nodes() {
+  CMD="${1}"
+  for node_ip in "${IP_VM01}" "${IP_VM02}" "${IP_VM03}"; do
+    ${SSH} ubuntu@${node_ip} ${CMD}
```

Comment:
Should we made this stricter? As in:
```
${SSH} ubuntu@${node_ip} ${CMD} || exit 1
```
and have the job fail if the command fails for whatever reason